group/assignment6/ParkingLot.java

```diff
@@ -26,6 +25,29 @@ public ParkingLot(int[] parkingLotCurrent, int[] parkingLotEnd) {
         carToPosition = new int[N - 1];
         this.parkingLotCurrent = parkingLotCurrent;
         moves = new ArrayList<>();
+        if (initialEqualsFinalFreeSpace(parkingLotCurrent, parkingLotEnd)) {
+        	for (int i = 0; i < N-1; i++) {
+                if (parkingLotCurrent[i] == -1) {
+                    if (i == 0) {
+                    	int counter = i+1;
+                    	while (counter<N-1 && parkingLotCurrent[counter] == parkingLotEnd[counter])
+                    	{counter++;}
+                    	if (counter!=N-1) {
+                    		int temp = parkingLotCurrent[i];
```

Comment:
If you are swapping cars, shouldn't these be added to the moves array?

Reply:
Why swap cars, which involves 2 moves, instead of moves only one car to the free space?